### ISSUE TITLE
onShown is not triggered

### ISSUE THREAD

**Issue by sherlock1982** [valor-software/ngx-bootstrap#1886]
onShown is not triggered in bsDropdown in 1.6.6
Was working in 1.6.1

**Comment by kraikill** [comment]
same problem for onHidden

**Comment by jmsmrgn** [comment]
same w/2.0.0-beta.2 and ng4

**Comment by valorkin** [comment]
I know where the issue is, I had to add inline show logic, and it leaves outside of component loader, so it doesn't triggers on shown and hide. Will be fixed

**Comment by mlamothe** [comment]
Do we have an ETA for a fix?

**Comment by valorkin** [comment]
I am on vacation till 24 of April, not sure I will get to PC next week

**Comment by mlamothe** [comment]
Since this broke our app, I put in a temporary workaround in our code like this:

```
private isOpenChange$: Subscription;
     menuIsopen = false;
    @ViewChild(BsDropdownDirective) dropdown;

    constructor(
        private authService: AuthenticationService,
        private busyService: BusyService,
        public userService: UserInfoService) {
    }

    ngOnInit(): void {
        // Workaround while onShown/onHidden are broken.
        this.isOpenChange$ = this.dropdown._state.isOpenChange.subscribe(isOpen => {
                    this.menuIsopen = isOpen;
        });
    }

    ngOnDestroy(): void {
        this.isOpenChange$.unsubscribe();
    }
```

**Comment by valorkin** [comment]
Sorry for inconvenience

**Comment by paesku** [comment]
@nwrdcore How did you make your code example work?

```javascript
import { Component, AfterViewInit, ViewChild } from '@angular/core';
import { BsDropdownDirective } from 'ngx-bootstrap';

@Component({
  selector: 'example-comp',
  templateUrl: `
<div class="btn-group" dropdown>
  <button dropdownToggle type="button" class="btn btn-primary dropdown-toggle">
    Button dropdown <span class="caret"></span>
  </button>
  <ul *dropdownMenu class="dropdown-menu" role="menu">
    <li role="menuitem"><a class="dropdown-item" href="#">Something else here</a></li>
    </li>
  </ul>
</div>
`
})
export class TopBarMenuComponent implements AfterViewInit {
  @ViewChild(BsDropdownDirective) dropdown: BsDropdownDirective;

  ngAfterViewInit() {
    console.log(this.dropdown);
  }
}
```

Logs undefined?

I can't access  **BsDropdownDirective**

**Comment by mlamothe** [comment]
Example HTML:
```
<div class="btn-group" dropdown placement="bottom right" (onHidden)="toggled(false)" (onShown)="toggled(true)" id="profile-menu">
  <button dropdownToggle type="button" class="btn btn-sm nav-bar-btn dropdown-toggle" [class.nav-bar-btn-open]="menuIsopen">

  </button>
  <ul *dropdownMenu class="dropdown-menu dropdown-menu-right user-menu" role="menu">

    <li role="menuitem" class="dropdown-item log-out">

    </li>
  </ul>
</div>
```

**Comment by mlamothe** [comment]
@paesku, honestly, I'm not sure - I'm not seeing any significant differences between my code and yours.

**Comment by paesku** [comment]
we are at angular 4.0.2. Other ngx components, like modal are working fine. weird

**Comment by ghost** [comment]
I have the same problem with the dropdown 

```
<div class="dropdown"
     dropdown
    (onHidden)="dropDownHide()">
   <ul>
      <li>
         <span>Test1</span>
      </li>
      <li>
         <span>Test2</span>
      </li>
      <li>
         <span>Test3</span>
      </li>
   </ul>
</div>
```

The function ```dropDownHide()``` is never triggered

**Comment by adurm-sw** [comment]
same issue here. Fortunately I was able to sort out a temporary (inefficient) workaround - but will change as soon as this is updated. Example below:

`@ViewChild(BsDropdownDirective) _dropDown : BsDropdownDirective;

....
ngDoCheck() {
    if(this._dropDown.isOpen) { doSomething() } 
  }
`

**Comment by ghost** [comment]
So there is no one looking to fix that ?

**Comment by valorkin** [comment]
https://github.com/valor-software/ngx-bootstrap/pull/1951/files

**Comment by ghost** [comment]
@valorkin if that pull request is the fix - when we could expect it in production ?

**Comment by keradus** [comment]
fix is merged, but not released :(
https://github.com/valor-software/ngx-bootstrap/compare/v1.6.6...development

**Comment by valorkin** [comment]
Fixing, one more issue, testing and publishing

**Comment by valorkin** [comment]
fix will be published in nearest hours as 1.7
